### PR TITLE
feat(appliance): deploy otel-collector

### PR DESCRIPTION
**Pull request by craigfurman**
**chore(appliance): name otel-agent things clearly**

Disambiguate agent from collector.

**chore(appliance): compare-helm can take multiple golden files**

Needed for otel, which is split into agent and collector in the
appliance but is represented by one component label in Helm.

**feat(appliance): deploy otel-collector**

---

Closes https://linear.app/sourcegraph/issue/REL-81/service-definition-otel-collector

## Test plan

Golden tests included.

## Changelog

- Appliance can deploy otel-collector

### CODE REVIEW

File: internal/appliance/reconciler/otel_collector_test.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package reconciler
+
+func (suite *ApplianceTestSuite) TestDeployOtelCollector() {
+	for _, tc := range []struct {
+		name string
+	}{
+		{name: "otel-collector/default"},
```
Comment (craigfurman) [inline internal/appliance/reconciler/otel_collector_test.go:7]:
```
go run ./internal/appliance/dev/compare-helm \
  -deploy-sourcegraph-helm-path ../../deploy-sourcegraph-helm \
  -component otel-collector \
  -golden-file internal/appliance/reconciler/testdata/golden-fixtures/otel-agent/default.yaml,internal/appliance/reconciler/testdata/golden-fixtures/otel-collector/default.yaml  \
  -helm-template-extra-args '--set openTelemetry.gateway.serviceAccount.create=true --set openTelemetry.agent.serviceAccount.create=true'

```

Nothing jumps off the page. Note that under this default config (no custom exporters, no jaeger), the configmap that is provisioned here but not by Helm, is not used.

```diff
2c2
< # helm: ServiceAccount/otel-agent
---
> # golden: ServiceAccount/otel-agent
5a6,8
>   annotations:
>     appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
>   creationTimestamp: "2024-04-19T00:00:00Z"
7,8d9
<     app.kubernetes.io/component: otel-collector
<     category: rbac
10a12,21
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>     - apiVersion: v1
>       blockOwnerDeletion: true
>       controller: true
>       kind: ConfigMap
>       name: sg
>       uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
12c23
< # helm: ServiceAccount/otel-collector
---
> # golden: ServiceAccount/otel-collector
15a27,29
>   annotations:
>     appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
>   creationTimestamp: "2024-04-19T00:00:00Z"
17,18d30
<     app.kubernetes.io/component: otel-collector
<     category: rbac
20a33,42
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>     - apiVersion: v1
>       blockOwnerDeletion: true
>       controller: true
>       kind: ConfigMap
>       name: sg
>       uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
22c44
< # helm: ConfigMap/otel-agent
---
> # golden: ConfigMap/otel-agent
68a91
> immutable: false
70a94,96
>   annotations:
>     appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
>   creationTimestamp: "2024-04-19T00:00:00Z"
72d97
<     app.kubernetes.io/component: otel-collector
74a100,109
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>     - apiVersion: v1
>       blockOwnerDeletion: true
>       controller: true
>       kind: ConfigMap
>       name: sg
>       uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
76c111
< # helm: Service/otel-collector
---
> # golden: Service/otel-collector
81,82c116,117
<     prometheus.io/port: "8888"
<     sourcegraph.prometheus/scrape: "true"
---
>     appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
>   creationTimestamp: "2024-04-19T00:00:00Z"
83a119
>     app: otel-collector
85,88d120
<     app.kubernetes.io/instance: release-name
<     app.kubernetes.io/managed-by: Helm
<     app.kubernetes.io/name: sourcegraph
<     app.kubernetes.io/version: 5.6.185
90d121
<     helm.sh/chart: sourcegraph-5.6.185
91a123,132
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>     - apiVersion: v1
>       blockOwnerDeletion: true
>       controller: true
>       kind: ConfigMap
>       name: sg
>       uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
92a134,140
>   clusterIP: NORMALIZED_FOR_TESTING
>   clusterIPs:
>     - NORMALIZED_FOR_TESTING
>   internalTrafficPolicy: Cluster
>   ipFamilies:
>     - IPv4
>   ipFamilyPolicy: SingleStack
103a152,153
>       protocol: TCP
>       targetPort: 8888
106,107c156
<     app.kubernetes.io/instance: release-name
<     app.kubernetes.io/name: sourcegraph
---
>   sessionAffinity: None
108a158,159
> status:
>   loadBalancer: {}
110c161
< # helm: DaemonSet/otel-agent
---
> # golden: DaemonSet/otel-agent
115c166,169
<     description: Forwards telemetry data to the OpenTelemetry Collector Deployment.
---
>     appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
>     deprecated.daemonset.template.generation: "1"
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   generation: 1
117,119c171
<     app.kubernetes.io/component: otel-collector
<     app.kubernetes.io/instance: release-name
<     app.kubernetes.io/managed-by: Helm
---
>     app.kubernetes.io/component: otel-agent
121c173
<     app.kubernetes.io/version: 5.6.185
---
>     app.kubernetes.io/version: 5.3.9104
123d174
<     helm.sh/chart: sourcegraph-5.6.185
124a176,185
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>     - apiVersion: v1
>       blockOwnerDeletion: true
>       controller: true
>       kind: ConfigMap
>       name: sg
>       uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
126c187,188
<   minReadySeconds: 5
---
>   minReadySeconds: 10
>   revisionHistoryLimit: 10
130,131d191
<       app.kubernetes.io/instance: release-name
<       app.kubernetes.io/name: sourcegraph
135a196
>       creationTimestamp: null
138,139d198
<         app.kubernetes.io/instance: release-name
<         app.kubernetes.io/name: sourcegraph
140a200
>       name: otel-agent
142d201
<       affinity: null
147,148c206
<           env: null
<           image: index.docker.io/sourcegraph/opentelemetry-collector:5.6.185@sha256:6d5fdb5bcadc518c2580662f28788b5866a48a848d2551d795221e49e1814452
---
>           image: index.docker.io/sourcegraph/opentelemetry-collector:5.3.9104
150a209
>             failureThreshold: 3
153a213,216
>               scheme: HTTP
>             periodSeconds: 10
>             successThreshold: 1
>             timeoutSeconds: 1
158a222
>               protocol: TCP
161c225,226
<               name: otlp-grpc
---
>               name: otel-grpc
>               protocol: TCP
164c229,230
<               name: otlp-http
---
>               name: otel-http
>               protocol: TCP
165a232
>             failureThreshold: 3
168a236,239
>               scheme: HTTP
>             periodSeconds: 10
>             successThreshold: 1
>             timeoutSeconds: 1
175a247,252
>           securityContext:
>             allowPrivilegeEscalation: false
>             readOnlyRootFilesystem: true
>             runAsGroup: 101
>             runAsUser: 100
>           terminationMessagePath: /dev/termination-log
180,183c257,265
<       nodeSelector: null
<       serviceAccountName: otel-agent
<       terminationGracePeriodSeconds: 120
<       tolerations: null
---
>       dnsPolicy: ClusterFirst
>       restartPolicy: Always
>       schedulerName: default-scheduler
>       securityContext:
>         fsGroup: 101
>         fsGroupChangePolicy: OnRootMismatch
>         runAsGroup: 101
>         runAsUser: 100
>       terminationGracePeriodSeconds: 30
185a268
>             defaultMode: 511
190a274,283
>   updateStrategy:
>     rollingUpdate:
>       maxSurge: 0
>       maxUnavailable: 1
>     type: RollingUpdate
> status:
>   currentNumberScheduled: 0
>   desiredNumberScheduled: 0
>   numberMisscheduled: 0
>   numberReady: 0
192c285
< # helm: Deployment/otel-collector
---
> # golden: Deployment/otel-collector
197c290,292
<     description: Receives, processes, and exports telemetry data.
---
>     appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   generation: 1
200,201d294
<     app.kubernetes.io/instance: release-name
<     app.kubernetes.io/managed-by: Helm
203c296
<     app.kubernetes.io/version: 5.6.185
---
>     app.kubernetes.io/version: 5.3.9104
205d297
<     helm.sh/chart: sourcegraph-5.6.185
206a299,308
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>     - apiVersion: v1
>       blockOwnerDeletion: true
>       controller: true
>       kind: ConfigMap
>       name: sg
>       uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
208,209c310,311
<   minReadySeconds: 5
<   progressDeadlineSeconds: 120
---
>   minReadySeconds: 10
>   progressDeadlineSeconds: 600
210a313
>   revisionHistoryLimit: 10
214,215c317,321
<       app.kubernetes.io/instance: release-name
<       app.kubernetes.io/name: sourcegraph
---
>   strategy:
>     rollingUpdate:
>       maxSurge: 25%
>       maxUnavailable: 25%
>     type: RollingUpdate
219a326
>       creationTimestamp: null
222,223d328
<         app.kubernetes.io/instance: release-name
<         app.kubernetes.io/name: sourcegraph
224a330
>       name: otel-collector
226d331
<       affinity: null
231,232c336
<           env: null
<           image: index.docker.io/sourcegraph/opentelemetry-collector:5.6.185@sha256:6d5fdb5bcadc518c2580662f28788b5866a48a848d2551d795221e49e1814452
---
>           image: index.docker.io/sourcegraph/opentelemetry-collector:5.3.9104
234a339
>             failureThreshold: 3
237a343,346
>               scheme: HTTP
>             periodSeconds: 10
>             successThreshold: 1
>             timeoutSeconds: 1
241a351
>               protocol: TCP
243a354
>               protocol: TCP
245a357
>               protocol: TCP
247a360
>               protocol: TCP
248a362
>             failureThreshold: 3
251a366,369
>               scheme: HTTP
>             periodSeconds: 10
>             successThreshold: 1
>             timeoutSeconds: 1
258a377,382
>           securityContext:
>             allowPrivilegeEscalation: false
>             readOnlyRootFilesystem: true
>             runAsGroup: 101
>             runAsUser: 100
>           terminationMessagePath: /dev/termination-log
260,261c384,392
<           volumeMounts: null
<       nodeSelector: null
---
>       dnsPolicy: ClusterFirst
>       restartPolicy: Always
>       schedulerName: default-scheduler
>       securityContext:
>         fsGroup: 101
>         fsGroupChangePolicy: OnRootMismatch
>         runAsGroup: 101
>         runAsUser: 100
>       serviceAccount: otel-collector
263,265c394,608
<       terminationGracePeriodSeconds: 120
<       tolerations: null
<       volumes: null
---
>       terminationGracePeriodSeconds: 30
> status: {}
> ---
> # golden: ConfigMap/sg
> apiVersion: v1
> data:
>   spec: |
>     spec:
>       requestedVersion: "5.3.9104"
> 
>       blobstore:
>         disabled: true
> 
>       codeInsights:
>         disabled: true
> 
>       codeIntel:
>         disabled: true
> 
>       frontend:
>         disabled: true
> 
>       grafana:
>         disabled: true
> 
>       gitServer:
>         disabled: true
> 
>       indexedSearch:
>         disabled: true
> 
>       nodeExporter:
>         disabled: true
> 
>       openTelemetryAgent: {}
> 
>       openTelemetryCollector:
>         disabled: true
> 
>       pgsql:
>         disabled: true
> 
>       postgresExporter:
>         disabled: true
> 
>       preciseCodeIntel:
>         disabled: true
> 
>       redisCache:
>         disabled: true
> 
>       redisExporter:
>         disabled: true
> 
>       redisStore:
>         disabled: true
> 
>       repoUpdater:
>         disabled: true
> 
>       searcher:
>         disabled: true
> 
>       symbols:
>         disabled: true
> 
>       syntectServer:
>         disabled: true
> 
>       worker:
>         disabled: true
> 
>       prometheus:
>         disabled: true
> kind: ConfigMap
> metadata:
>   annotations:
>     appliance.sourcegraph.com/currentVersion: 5.3.9104
>     appliance.sourcegraph.com/managed: "true"
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   name: sg
>   namespace: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
> ---
> # golden: ConfigMap/otel-collector
> apiVersion: v1
> data:
>   config.yaml: |
>     receivers:
>       otlp:
>         protocols:
>           grpc: # port 4317
>           http: # port 4318
> 
>     exporters:
>           {}
> 
> 
>     extensions:
>       health_check:
>         endpoint: ":13133"
>       zpages:
>         endpoint: "localhost:55679"
> 
>     service:
>       extensions:
>         - health_check
>         - zpages
>       pipelines:
>         traces:
>           receivers:
>             - otlp
>           exporters:
> immutable: false
> kind: ConfigMap
> metadata:
>   annotations:
>     appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   labels:
>     deploy: sourcegraph
>   name: otel-collector
>   namespace: NORMALIZED_FOR_TESTING
>   ownerReferences:
>     - apiVersion: v1
>       blockOwnerDeletion: true
>       controller: true
>       kind: ConfigMap
>       name: sg
>       uid: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING
> ---
> # golden: ConfigMap/sg
> apiVersion: v1
> data:
>   spec: |
>     spec:
>       requestedVersion: "5.3.9104"
> 
>       blobstore:
>         disabled: true
> 
>       codeInsights:
>         disabled: true
> 
>       codeIntel:
>         disabled: true
> 
>       frontend:
>         disabled: true
> 
>       grafana:
>         disabled: true
> 
>       gitServer:
>         disabled: true
> 
>       indexedSearch:
>         disabled: true
> 
>       nodeExporter:
>         disabled: true
> 
>       openTelemetryAgent:
>         disabled: true
> 
>       openTelemetryCollector: {}
> 
>       pgsql:
>         disabled: true
> 
>       postgresExporter:
>         disabled: true
> 
>       preciseCodeIntel:
>         disabled: true
> 
>       redisCache:
>         disabled: true
> 
>       redisExporter:
>         disabled: true
> 
>       redisStore:
>         disabled: true
> 
>       repoUpdater:
>         disabled: true
> 
>       searcher:
>         disabled: true
> 
>       symbols:
>         disabled: true
> 
>       syntectServer:
>         disabled: true
> 
>       worker:
>         disabled: true
> 
>       prometheus:
>         disabled: true
> kind: ConfigMap
> metadata:
>   annotations:
>     appliance.sourcegraph.com/currentVersion: 5.3.9104
>     appliance.sourcegraph.com/managed: "true"
>   creationTimestamp: "2024-04-19T00:00:00Z"
>   name: sg
>   namespace: NORMALIZED_FOR_TESTING
>   resourceVersion: NORMALIZED_FOR_TESTING
>   uid: NORMALIZED_FOR_TESTING

```

File: internal/appliance/reconciler/otel_collector_test.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+		{name: "otel-collector/with-exporters"},
```
Comment (craigfurman) [inline internal/appliance/reconciler/otel_collector_test.go:8]: `❯ diff internal/appliance/reconciler/testdata/golden-fixtures/otel-collector/default.yaml internal/appliance/reconciler/testdata/golden-fixtures/otel-collector/with-exporters.yaml`

```diff
6c6
<         appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
---
>         appliance.sourcegraph.com/configHash: b4fcee234558123758f5b40cd65465d1dc69f0a46be2f26d404c039c41ec6358
51c51
<                 - --config=/etc/otel-collector/configs/logging.yaml
---
>                 - --config=/etc/otel-collector/config.yaml
99a100,102
>               volumeMounts:
>                 - mountPath: /etc/otel-collector
>                   name: config
110a114,121
>           volumes:
>             - configMap:
>                 defaultMode: 511
>                 items:
>                   - key: config.yaml
>                     path: config.yaml
>                 name: otel-collector
>               name: config
122c133,135
<               {}
---
>               an-exporter:
>                   some-config:
>                       foo: bar
139a153
>                 - an-exporter
144c158
<         appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
---
>         appliance.sourcegraph.com/configHash: b4fcee234558123758f5b40cd65465d1dc69f0a46be2f26d404c039c41ec6358
192c206,210
<           openTelemetryCollector: {}
---
>           openTelemetryCollector:
>             exporters:
>               an-exporter:
>                 some-config:
>                   foo: bar
243c261
<         appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
---
>         appliance.sourcegraph.com/configHash: b4fcee234558123758f5b40cd65465d1dc69f0a46be2f26d404c039c41ec6358
262c280
<         appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
---
>         appliance.sourcegraph.com/configHash: b4fcee234558123758f5b40cd65465d1dc69f0a46be2f26d404c039c41ec6358

```

File: internal/appliance/reconciler/otel_collector_test.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+		{name: "otel-collector/with-exporters-tls-secret-name"},
```
Comment (craigfurman) [inline internal/appliance/reconciler/otel_collector_test.go:9]: `❯ diff internal/appliance/reconciler/testdata/golden-fixtures/otel-collector/default.yaml internal/appliance/reconciler/testdata/golden-fixtures/otel-collector/with-exporters-tls-secret-name.yaml`

```diff
6c6
<         appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
---
>         appliance.sourcegraph.com/configHash: c6eb914ad149f9d61ad417958f0e6d227ffb812ecdef4f29a4db4ad3a82af831
99a100,102
>               volumeMounts:
>                 - mountPath: /tls
>                   name: otel-collector-tls
110a114,118
>           volumes:
>             - name: otel-collector-tls
>               secret:
>                 defaultMode: 420
>                 secretName: a-secret
144c152
<         appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
---
>         appliance.sourcegraph.com/configHash: c6eb914ad149f9d61ad417958f0e6d227ffb812ecdef4f29a4db4ad3a82af831
192c200,201
<           openTelemetryCollector: {}
---
>           openTelemetryCollector:
>             exportersTlsSecretName: a-secret
243c252
<         appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
---
>         appliance.sourcegraph.com/configHash: c6eb914ad149f9d61ad417958f0e6d227ffb812ecdef4f29a4db4ad3a82af831
262c271
<         appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
---
>         appliance.sourcegraph.com/configHash: c6eb914ad149f9d61ad417958f0e6d227ffb812ecdef4f29a4db4ad3a82af831

```

File: internal/appliance/reconciler/otel_collector_test.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+		{name: "otel-collector/with-jaeger"},
```
Comment (craigfurman) [inline internal/appliance/reconciler/otel_collector_test.go:10]: `❯ diff internal/appliance/reconciler/testdata/golden-fixtures/otel-collector/default.yaml internal/appliance/reconciler/testdata/golden-fixtures/otel-collector/with-jaeger.yaml`

Mostly noise from the added jaeger resources, but note the extra env vars and different config file in the container command.

```diff
5a6,104
>         appliance.sourcegraph.com/configHash: dea9f009a80b7d3f307e194f460f8d5b78deb6f565e683e7d6253a702b3e1912
>       creationTimestamp: "2024-04-19T00:00:00Z"
>       generation: 1
>       labels:
>         app.kubernetes.io/component: jaeger
>         app.kubernetes.io/name: sourcegraph
>         app.kubernetes.io/version: 5.3.9104
>         deploy: sourcegraph
>       name: jaeger
>       namespace: NORMALIZED_FOR_TESTING
>       ownerReferences:
>         - apiVersion: v1
>           blockOwnerDeletion: true
>           controller: true
>           kind: ConfigMap
>           name: sg
>           uid: NORMALIZED_FOR_TESTING
>       resourceVersion: NORMALIZED_FOR_TESTING
>       uid: NORMALIZED_FOR_TESTING
>     spec:
>       minReadySeconds: 10
>       progressDeadlineSeconds: 600
>       replicas: 1
>       revisionHistoryLimit: 10
>       selector:
>         matchLabels:
>           app: jaeger
>       strategy:
>         type: Recreate
>       template:
>         metadata:
>           annotations:
>             prometheus.io/port: "16686"
>             prometheus.io/scrape: "true"
>           creationTimestamp: null
>           labels:
>             app: jaeger
>             deploy: sourcegraph
>           name: jaeger
>         spec:
>           containers:
>             - args:
>                 - --memory.max-traces=20000
>               image: index.docker.io/sourcegraph/jaeger-all-in-one:5.3.9104
>               imagePullPolicy: IfNotPresent
>               name: jaeger
>               ports:
>                 - containerPort: 5775
>                   protocol: UDP
>                 - containerPort: 6831
>                   protocol: UDP
>                 - containerPort: 6832
>                   protocol: UDP
>                 - containerPort: 5778
>                   protocol: TCP
>                 - containerPort: 16686
>                   protocol: TCP
>                 - containerPort: 14250
>                   protocol: TCP
>               readinessProbe:
>                 failureThreshold: 3
>                 httpGet:
>                   path: /
>                   port: 14269
>                   scheme: HTTP
>                 initialDelaySeconds: 5
>                 periodSeconds: 10
>                 successThreshold: 1
>                 timeoutSeconds: 1
>               resources:
>                 limits:
>                   cpu: "1"
>                   memory: 1G
>                 requests:
>                   cpu: 500m
>                   memory: 500M
>               securityContext:
>                 allowPrivilegeEscalation: false
>                 readOnlyRootFilesystem: true
>                 runAsGroup: 101
>                 runAsUser: 100
>               terminationMessagePath: /dev/termination-log
>               terminationMessagePolicy: FallbackToLogsOnError
>           dnsPolicy: ClusterFirst
>           restartPolicy: Always
>           schedulerName: default-scheduler
>           securityContext:
>             fsGroup: 101
>             fsGroupChangePolicy: OnRootMismatch
>             runAsGroup: 101
>             runAsUser: 100
>           serviceAccount: jaeger
>           serviceAccountName: jaeger
>           terminationGracePeriodSeconds: 30
>     status: {}
>   - apiVersion: apps/v1
>     kind: Deployment
>     metadata:
>       annotations:
51c150,157
<                 - --config=/etc/otel-collector/configs/logging.yaml
---
>                 - --config=/etc/otel-collector/configs/jaeger.yaml
>               env:
>                 - name: JAEGER_HOST
>                   value: jaeger-collector
>                 - name: JAEGER_OTLP_GRPC_PORT
>                   value: "4320"
>                 - name: JAEGER_OTLP_HTTP_PORT
>                   value: "4321"
114,160d219
<       config.yaml: |
<         receivers:
<           otlp:
<             protocols:
<               grpc: # port 4317
<               http: # port 4318
< 
<         exporters:
<               {}
< 
< 
<         extensions:
<           health_check:
<             endpoint: ":13133"
<           zpages:
<             endpoint: "localhost:55679"
< 
<         service:
<           extensions:
<             - health_check
<             - zpages
<           pipelines:
<             traces:
<               receivers:
<                 - otlp
<               exporters:
<     immutable: false
<     kind: ConfigMap
<     metadata:
<       annotations:
<         appliance.sourcegraph.com/configHash: c84ac276e60e85a888146303bb5d592cb42507dc290a515051c6ecedc03d6f4c
<       creationTimestamp: "2024-04-19T00:00:00Z"
<       labels:
<         deploy: sourcegraph
<       name: otel-collector
<       namespace: NORMALIZED_FOR_TESTING
<       ownerReferences:
<         - apiVersion: v1
<           blockOwnerDeletion: true
<           controller: true
<           kind: ConfigMap
<           name: sg
<           uid: NORMALIZED_FOR_TESTING
<       resourceVersion: NORMALIZED_FOR_TESTING
<       uid: NORMALIZED_FOR_TESTING
<   - apiVersion: v1
<     data:
193a253,255
>           jaeger:
>             disabled: false
> 
242a305,323
>         appliance.sourcegraph.com/configHash: dea9f009a80b7d3f307e194f460f8d5b78deb6f565e683e7d6253a702b3e1912
>       creationTimestamp: "2024-04-19T00:00:00Z"
>       labels:
>         deploy: sourcegraph
>       name: jaeger
>       namespace: NORMALIZED_FOR_TESTING
>       ownerReferences:
>         - apiVersion: v1
>           blockOwnerDeletion: true
>           controller: true
>           kind: ConfigMap
>           name: sg
>           uid: NORMALIZED_FOR_TESTING
>       resourceVersion: NORMALIZED_FOR_TESTING
>       uid: NORMALIZED_FOR_TESTING
>   - apiVersion: v1
>     kind: ServiceAccount
>     metadata:
>       annotations:
257a339,428
>   - apiVersion: v1
>     kind: Service
>     metadata:
>       annotations:
>         appliance.sourcegraph.com/configHash: dea9f009a80b7d3f307e194f460f8d5b78deb6f565e683e7d6253a702b3e1912
>       creationTimestamp: "2024-04-19T00:00:00Z"
>       labels:
>         app: jaeger-collector
>         app.kubernetes.io/component: jaeger-collector
>         deploy: sourcegraph
>       name: jaeger-collector
>       namespace: NORMALIZED_FOR_TESTING
>       ownerReferences:
>         - apiVersion: v1
>           blockOwnerDeletion: true
>           controller: true
>           kind: ConfigMap
>           name: sg
>           uid: NORMALIZED_FOR_TESTING
>       resourceVersion: NORMALIZED_FOR_TESTING
>       uid: NORMALIZED_FOR_TESTING
>     spec:
>       clusterIP: NORMALIZED_FOR_TESTING
>       clusterIPs:
>         - NORMALIZED_FOR_TESTING
>       internalTrafficPolicy: Cluster
>       ipFamilies:
>         - IPv4
>       ipFamilyPolicy: SingleStack
>       ports:
>         - name: jaeger-collector-tchannel
>           port: 14267
>           protocol: TCP
>           targetPort: 14267
>         - name: jaeger-collector-http
>           port: 14268
>           protocol: TCP
>           targetPort: 14268
>         - name: jaeger-collector-grpc
>           port: 14250
>           protocol: TCP
>           targetPort: 14250
>       selector:
>         app: jaeger
>         app.kubernetes.io/component: all-in-one
>       sessionAffinity: None
>       type: ClusterIP
>     status:
>       loadBalancer: {}
>   - apiVersion: v1
>     kind: Service
>     metadata:
>       annotations:
>         appliance.sourcegraph.com/configHash: dea9f009a80b7d3f307e194f460f8d5b78deb6f565e683e7d6253a702b3e1912
>       creationTimestamp: "2024-04-19T00:00:00Z"
>       labels:
>         app: jaeger-query
>         app.kubernetes.io/component: jaeger-query
>         deploy: sourcegraph
>       name: jaeger-query
>       namespace: NORMALIZED_FOR_TESTING
>       ownerReferences:
>         - apiVersion: v1
>           blockOwnerDeletion: true
>           controller: true
>           kind: ConfigMap
>           name: sg
>           uid: NORMALIZED_FOR_TESTING
>       resourceVersion: NORMALIZED_FOR_TESTING
>       uid: NORMALIZED_FOR_TESTING
>     spec:
>       clusterIP: NORMALIZED_FOR_TESTING
>       clusterIPs:
>         - NORMALIZED_FOR_TESTING
>       internalTrafficPolicy: Cluster
>       ipFamilies:
>         - IPv4
>       ipFamilyPolicy: SingleStack
>       ports:
>         - name: query-http
>           port: 16686
>           protocol: TCP
>           targetPort: 16686
>       selector:
>         app: jaeger
>         app.kubernetes.io/component: all-in-one
>       sessionAffinity: None
>       type: ClusterIP
>     status:
>       loadBalancer: {}

```

**Review by craigfurman — COMMENTED** [review]
I've tried to mimic all the conditionals in the helm chart. This was a relatively complex one.

**Review by DaedalusG — APPROVED** [review]
Are the golden tests still in `5.3.9104` ?

**Comment by craigfurman** [comment]
@DaedalusG yeah they are, but I forgot to mention: the compare-helm diffs in here are against main of the helm chart, not 5.3.9104. Some resource config had changed so I picked the up-to-date one.

We could update all the golden fixtures, and then do a big compare-helm for all services to check for any config drift. That's basically what https://linear.app/sourcegraph/issue/REL-122/run-compare-helm-against-all-services-and-squash-any-problematic is IIRC.

Bonus points for extracting the versions out of all sg input fixtures into one place, so that we don't have to update it with shotgun surgery using xargs and sed 😂

**Pull request merged**